curl/src/main/scala/org/http4s/curl/websocket/Error.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.curl.websocket
+
+sealed trait Error extends Throwable with Serializable with Product
```

Comment:
Directly extending `Throwable` seems to be discouraged. Also since these are `object` should they extend `NoStackTrace` ?

https://contributors.scala-lang.org/t/nonfatal-and-controlthrowable-changed-in-stdlib/6049/5

```suggestion
sealed trait Error extends Exception with Serializable with Product
```